src/test/shell/bazel/bazel_rules_test.sh

```diff
@@ -743,4 +743,153 @@ EOF
     bazel run //pkg:my_executable >$TEST_log 2>&1 || fail "Binary should have exit code 0"
 }
 
+function test_starlark_rule_with_run_under_env() {
+  mkdir pkg
+  cat >pkg/BUILD <<'EOF'
+load(":rules.bzl", "my_executable", "my_test", "my_wrapper")
+my_executable(
+  name = "my_executable",
+)
+my_test(
+  name = "my_test",
+)
+my_wrapper(
+  name = "my_wrapper",
+)
+EOF
+
+  # On Windows this file needs to be acceptable by CreateProcessW(), rather
+  # than a Bourne script.
+  if "$is_windows"; then
+    local run_under_script="pkg/run_under.bat"
+    echo 'exports_files(["run_under.bat"])' >> pkg/BUILD
+    cat >pkg/run_under.bat <<'EOF'
+@ECHO OFF
+if not defined RUN_UNDER_VALUE (
+  echo --run_under script was unexpectedly executed
+)
+exit /B 101
+EOF
+    cat >pkg/rules.bzl <<'EOF'
+_SCRIPT_EXT = ".bat"
+_SCRIPT_CONTENT = """
+if not defined RUN_UNDER_VALUE (
+  echo run_under_env value is not set
+  exit /B 1
+)
+start /b /wait bash.exe -c "%RUN_UNDER_VALUE%"
+if %ERRORLEVEL% NEQ 101 (
+  echo Failed to execute run_under_env value
+  exit /B 1
+)
+exit /B 0
+"""
+EOF
+  else
+  local run_under_script="pkg/run_under.sh"
+  echo 'exports_files(["run_under.sh"])' >> pkg/BUILD
+    cat >pkg/run_under.sh <<'EOF'
+#!/bin/bash
+if [[ -z "${RUN_UNDER_VALUE}" ]]; then
+  echo --run_under script was unexpectedly executed
+fi
+exit 101
+EOF
+    chmod +x pkg/run_under.sh
+    cat >pkg/rules.bzl <<'EOF'
+_SCRIPT_EXT = ".sh"
+_SCRIPT_CONTENT = """#!/bin/bash
+if [[ -z "${RUN_UNDER_VALUE:-}" ]]; then
+  echo "run_under_env value is not set"
+  exit 1
+fi
+${RUN_UNDER_VALUE}
+if [[ $? != 101 ]]; then
+  echo "Failed to execute run_under_env value"
+  exit 1
+fi
+exit 0
+"""
+EOF
+  fi
+
+  cat >>pkg/rules.bzl <<'EOF'
+def _my_executable_impl(ctx):
+    executable_sh = ctx.actions.declare_file(ctx.attr.name + _SCRIPT_EXT)
+    ctx.actions.write(
+        output = executable_sh,
+        content = _SCRIPT_CONTENT,
+        is_executable = True,
+    )
+    return [
+        DefaultInfo(
+            executable = executable_sh,
+        ),
+        RunEnvironmentInfo(
+            run_under_env = "RUN_UNDER_VALUE",
+        ),
+    ]
+
+my_executable = rule(
+    implementation = _my_executable_impl,
+    attrs = {},
+    executable = True,
+)
+
+my_test = rule(
+    implementation = _my_executable_impl,
+    attrs = {},
+    test = True,
+)
+
+def _my_wrapper_impl(ctx):
+    executable_sh = ctx.actions.declare_file(ctx.attr.name + _SCRIPT_EXT)
+    ctx.actions.expand_template(
+        output = executable_sh,
+        template = ctx.file._script,
+        substitutions = {},
+        is_executable = True,
+    )
+    return [DefaultInfo(
+        executable = executable_sh,
+    )]
+
+my_wrapper = rule(
+    implementation = _my_wrapper_impl,
+    attrs = {
+        "_script": attr.label(
+            allow_single_file = True,
+            default = Label("//pkg:run_under" + _SCRIPT_EXT),
```

Comment:
The `Label(...)` call isn't needed, you can just pass a string.

Reply:
Sure, though in general I would expect all default values for rules to use `Label(...)` since it offers more consistent results across repository boundaries. It's not strictly needed for this test but would be a better thing for other devs to pattern match.